src/main/js/apiclient.js

```diff
@@ -52,6 +52,17 @@ class ApiClient {
             .then(this.handleCasAuthentication);
     }
 
+    delete(url: string, payload: any) {
+        const deleteOptions = {
+            method: 'DELETE',
+            body: JSON.stringify(payload),
+        };
+        const options = Object.assign(deleteOptions, fetchOptions);
+        options.headers['Content-Type'] = 'application/json';
+        return fetch(createUrl(url), options)
+            .then(this.handleCasAuthentication);
+    }
+
```

Comment:
It looks like an exact copy of post. Please create a generic function and call it from post and delete.